### PR TITLE
Add HTTP protocol test case to OTLP exporter integration tests

### PR DESCRIPTION
**Pull request by pellared**
Add HTTP protocol test case to OTLP exporter integration tests to help with https://github.com/open-telemetry/opentelemetry-dotnet/pull/2691

## Remarks

```
AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport", true);
```

is only needed for the gRPC OTLP protocol (on .NET 3.1 Core of course).

### CODE REVIEW

**Comment by tomkerkhove** [comment]
I don't see it either so it might be the collector :/

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTests.cs [hunk 1]
```diff
@@ -24,23 +24,26 @@ namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests
...
+            // Adding the OtlpExporter creates a GrpcChannel/HttpClient.
+            // This switch must be set before creating a GrpcChannel/HttpClient when calling an insecure HTTP/2 endpoint.
             // See: https://docs.microsoft.com/aspnet/core/grpc/troubleshoot#call-insecure-grpc-services-with-net-core-client
             AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport", true);
 #endif
 
             var exporterOptions = new OtlpExporterOptions
             {
-                Endpoint = new System.Uri($"http://{CollectorEndpoint}"),
+                Endpoint = new System.Uri($"http://{CollectorEndpoint}:{port}"),
```
Comment (reyang) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTests.cs:45]: Not introduced by this PR, the term `CollectorEndpoint` seems to be confusing - perhaps endpoint should include the port?

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTests.cs [hunk 2]
```diff
@@ -24,23 +24,26 @@ namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests
 {
     public class IntegrationTests
     {
-        private const string CollectorEndpointEnvVarName = "OTEL_EXPORTER_OTLP_ENDPOINT";
-        private static readonly string CollectorEndpoint = SkipUnlessEnvVarFoundFactAttribute.GetEnvironmentVariable(CollectorEndpointEnvVarName);
+        private const string CollectorEndpointEnvVarName = "OTEL_COLLECTOR_HOSTNAME";
```
Comment (reyang) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTests.cs:27]: Curious, what's the different between endpoint and hostname? (my understanding is that TCP endpoint contains more information than a hostname - based on experience working with C++ Boost https://www.boost.org/doc/libs/1_47_0/doc/html/boost_asio/reference/ip__tcp/endpoint.html)

**Review by reyang — APPROVED** [review]
LGTM in general, perhaps we want to rename `CollectorEndpoint` to `CollectorHostname`/`CollectorHostName`.

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTests.cs (hunk 1 above)
Comment (pellared) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTests.cs:45]: CollectorHostname will be better

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTests.cs (hunk 2 above)
Comment (pellared) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTests.cs:27]: That is exactly how I see it

**Comment by pellared** [comment]
> LGTM in general, perhaps we want to rename `CollectorEndpoint` to `CollectorHostname`/`CollectorHostName`.

Agree. The main problem is.... That the test for HttpProtobuf are failing and the reason is unknown. I do not know when I will have time to investigate it further

**Comment by reyang** [comment]
> > LGTM in general, perhaps we want to rename `CollectorEndpoint` to `CollectorHostname`/`CollectorHostName`.
> 
> Agree. The main problem is.... That the test for HttpProtobuf are failing and the reason is unknown. I do not know when I will have time to investigate it further

Is this a priority? I don't know much history about this component, why would people want to use HTTP instead of gRPC? I can imagine the following:
1. Work with HTTP proxies, so it's simpler to penetrate firewalls.
2. Required by some company policies.
3. Less dependency? (I doubt if is this the case today)

**Comment by pellared** [comment]
I heard that it is also more performant and it is the default proposed by spec https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md.

I think the URL for HttpProtobuf is bad in the test code. I will check it on Monday

**Comment by pellared** [comment]
@tomkerkhove There were two issues:
1. a bug in test (missing whole trace endpoint URL)
2. For an unknown reason the collector was not listening on the default HTTP endpoint port at all. I had to explicitly set the endpoint URL in the config. When I have more time I'll check if it is already fixed in https://github.com/open-telemetry/opentelemetry-collector

**Comment by pellared** [comment]
@open-telemetry/dotnet-approvers PTAL

**Comment by tomkerkhove** [comment]
Thanks for sharing and taking a closer look! I've noticed you have removed the HTTP comment remark, is that on purpose?

**Comment by pellared** [comment]
> Thanks for sharing and taking a closer look! I've noticed you have removed the HTTP comment remark, is that on purpose?

Yes, because it is not needed and the integration test confirms it. Take notice:

```csharp
#if NETCOREAPP3_1
            // Adding the OtlpExporter creates a GrpcChannel.
            // This switch must be set before creating a GrpcChannel when calling an insecure HTTP/2 endpoint.
            // See: https://docs.microsoft.com/aspnet/core/grpc/troubleshoot#call-insecure-grpc-services-with-net-core-client
            if (protocol == OtlpExportProtocol.Grpc)
            {
                AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport", true);
            }
#endif
```

**Review by alanwest — APPROVED** [review]
LGTM

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
> > > LGTM in general, perhaps we want to rename `CollectorEndpoint` to `CollectorHostname`/`CollectorHostName`.
> > 
> > 
> > Agree. The main problem is.... That the test for HttpProtobuf are failing and the reason is unknown. I do not know when I will have time to investigate it further
> 
> Is this a priority? I don't know much history about this component, why would people want to use HTTP instead of gRPC? I can imagine the following:
> 
> 1. Work with HTTP proxies, so it's simpler to penetrate firewalls.
> 2. Required by some company policies.
> 3. Less dependency? (I doubt if is this the case today)

^ There are backends which doesn't support Grpc.

**Pull request merged**